tool/tsh/db.go

```diff
@@ -1144,6 +1170,10 @@ func withConnectRequirements(ctx context.Context, tc *client.TeleportClient, rou
 			// via `tsh db connect`.
 			// (elasticsearch-sql-cli cannot be configured to use specific certs).
 			r.addLocalProxyWithTunnel(formatDBProtocolReason(route.Protocol))
+		case defaults.ProtocolPostgres:
+			// postgres needs a local proxy to support cancel requests.
+			r.addLocalProxy(fmt.Sprintf("%v and cancel requests require local proxy",
+				formatDBProtocolReason(route.Protocol)))
```

Comment:
note that this is in `withConnectRequirements`, as in `tsh db connect`. So this will only ensure we start  a local proxy with `tsh db connect`.

Reply:
is that ok?

Reply:
> So this will only ensure we start a local proxy with tsh db connect.
Even if this is only `tsh db connect` the default behaviour was changed. If a customer was using Separate listener mode the `tsh db connect`  flow was not using local proxy. After that change if cluster was configured with separate_listener_mode  the `tsh db connect` will trigger TLS Routing flow. 
My main point is not to change this behaviour.

Reply:
Is the concern here due to local proxy causing TLS-inside-TLS behavior? Because without this change, `tsh db connect` will not support cancel requests in separate listener mode.

Reply:
> Is the concern here due to local proxy causing TLS-inside-TLS behavior? 
I don't that that TLS inside TLS is  our concern. We are using this behavior in other integration but we don't want to change/alter suddenly current behavior and make the `--tunnel` mode as default for all postgres connection. 
  
Why the proxy cannot run without --tunnel mode for normal connection and upgrade a new connection to TLS only if the cancel request was detected ?

Reply:
@GavinFrazar 
Even if we upgrade the connection only to TLS without `--tunnel` option this behaviour can break the client integration. 

Let's imagine that the client is using the following sperate teleport proxy configuration: 
```
auth_service:
  proxy_listener_mode: "separate"
  
proxy_service:
  enabled: "true"
  public_addr:  ['alb.master.ice-berg.dev:3080']
  ssh_public_addr: ['master.ice-berg.dev:3023']
  tunnel_public_addr: ['master.ice-berg.dev:3024']
  postgres_public_addr: ['master.ice-berg.dev:5432']
  web_listen_addr: 0.0.0.0:3080
  postgres_listen_addr: 0.0.0.0:5432
```

where `alb.master.ice-berg.dev:`   is an L7 Load Balancer like AWS ALB that terminates TLS connection and dials the teleport proxy without forwarding TLS extensions. 

before this change the `tsh db connect postgres` command was using the `master.ice-berg.dev:3024` address after switching to a local proxy for postgres connection the `tsh db connect postgres` will try to connect to `alb.master.ice-berg.dev:3080` with TLS Routing extension. Due to L7 ALB the TLS exertion will be dropped and the client will not be able to connect to the right service. Recently this case was fixed by adding support for connection upgrade but I can still imagine that the v13 client can be used to connect to the old teleport proxy that doesn't support [Connection Upgrade](https://github.com/gravitational/teleport/issues/14369) feature.

Reply:
Hey guys. I know we have code to track what mode is required, when, which protocol, etc.. but I've decided to create a spreadsheet to track it as it's slightly more visual. Feel free to update it or fix it if something is not accurate:
https://docs.google.com/spreadsheets/d/1m0swp5F4PZ9-C6pAUR45FbIB1o2IeJ3mjmJz1VrRG3g/edit?usp=sharing